projects/components/src/table/table.component.ts

```diff
@@ -278,6 +276,15 @@ export class TableComponent
       this.columnConfigsSubject.next(this.buildColumnConfigs());
     }
 
+    if (changes.selections) {
+      this.dataSource?.unselectAllRows();
+      this.selections?.forEach(row => {
```

Comment:
What's the reasoning for unselecting all rows and explicitly setting them again, rather than modifying only the changed rows? It's not clear to me, so probably worth a comment (or code change)

Reply:
We need to unselect the rows which were already checked before, if there is a change in the selections

That can either be done by maintaining a state of selected rows, or unselecting everything before and then selecting the newly selected ones

The current piece of code looks cleaner to me, instead of maintaining a state of selected rows

Reply:
would `changes.selections?.previousValue` suffice?

Reply:
Tried that. But, there is something which needs to go through data source using `cachedRows`. Couldn't seem to work around that